### ISSUE TITLE
DSUBm_005: Should $events operation be required

### ISSUE THREAD

**Issue by scontento** [IHE/ITI.DSUBm#13]
(see DSUBm_005 issues.html)
The [Resource Subscription Search [ITI-113]](https://profiles.ihe.net/ITI/DSUBm/1.0.0-comment/ITI-113.html) define the $events operation that permits to retrieve from the Resource Notification Broker the events associated to a Subscription. This operation could be use to retrieve puntually the missed event when, for example, a connection problem occur and the Resource Notification Recipient does note receive the notification. But, this operation requires the Resource Notification Broker to be capable of maintain previous events associated with the Subscriptions. Should the supporting of this operation be required for the Resource Notification Broker or should be maintain optional?

**Comment by RicardoQuintano** [comment]
I think it is important to set the _$events_ operation as REQUIRED. 

The fact of a Recipient NOT receiving a notification (e.g., due to connection problems) may result in a Task not done by the clinical staff that might reflect in bad consequences for the patient (e.g., not getting a medication because the physician was not notified that lab tests where ready).

**Comment by GinoCanessa** [comment]
I agree that `$events` needs to be required if the channel is not "reliable" (guaranteed delivery).  Given that DSUBm prefers `rest-hook`, there needs to be some way for clients to recover in the case of errors.

**Comment by scontento** [comment]
Leave this as a open issue for discussion. 

Leaving the operation as optional let this operation is still available if implemented. (maybe a reccomendation could be useful to the reader/implementers?)

Considering the requirement of this operation implies that the Broker shall keep track of the history of the events that have occurred. 
Is this consideration relevant to the decision on this operation?
